docs/puzzles/day17.md

```diff
@@ -2,10 +2,410 @@ import Solver from "../../../../website/src/components/Solver.js"
+    Velocity(x = vel.x - vel.x.sign, y = vel.y - 1)
+  )
+```
+
+#### A Target Area
+
+Next we are told that a successful launch trajectory will cause the probe
+to be within in a target area after at least one of its steps. The area is
+defined by the points within a given range in the `x` and `y` directions.
+We can model a target area by a case class with two ranges:
+```scala
+case class Target(xs: Range, ys: Range)
+```
+
+### Reasoning about the Problem
+
+We are told to first identify the initial velocity of trajectories that will
+hit the target area exactly, meaning that after a given step, the probe will
+be exactly within the target area; and that we must disregard trajectories
+that "overshoot" i.e. pass through the entire target area in a single step.
+
+We are also told to find the maximum height (`y` position) reached out of
+all valid trajectories.
+
+#### Simulating a trajectory
+
+For this problem we will simulate a probe moving along a trajectory, i.e.
+iterate every step of the trajectory until the probe either collides with
+the target or has moved beyond it. And at each step we will record if the
+current height of the probe is higher than before.
+
+##### Checking Collisions
+To identify if the probe collides with the target, we check that its
+`x` position is within the `xs` range of the target and also that its
+`y` position is within the `ys` range of the target:
+```scala
+def collides(probe: Probe, target: Target): Boolean =
+  val Probe(pos, _) = probe
+  val Target(xs, ys) = target
+  xs.contains(pos.x) && ys.contains(pos.y)
+```
+
+##### Has the Probe Moved Beyond the Target?
+We can check that the probe has moved beyond the target by considering
+situations for each direction:
+- for the `x` direction:
+  - the `x` velocity is `0` and the `x` position of the probe is less than
+    the minimum `x` position of the target,
+  - the `x` position of the probe is greater than the maximum `x` position
+    of the target.
+- for the `y` direction:
+  - the `y` velocity is less than `0` and the
+    `y` position of the probe is less than the minimum `y` position of the
+    target.
+
+The code to compute this is given as such:
+```scala
+def beyond(probe: Probe, target: Target): Boolean =
+  val Probe(pos, vel) = probe
+  val Target(xs, ys) = target
+  val beyondX = (vel.x == 0 && pos.x < xs.min) || pos.x > xs.max
+  val beyondY = vel.y < 0 && pos.y < ys.min
+  beyondX || beyondY
+```
+
+:::info
+The above conditions make the assumptions that the `x` velocity is never
+negative, and that the target is always in the positive `x` direction.
+They are also informed by the fact that the probe will eventually always
+have negative velocity (due to gravity).
+:::
+
+##### Running the Simulation
+
+We can use our two conditions to now simulate the trajectory of a probe:
+- We begin with an initial `probe`, and an initial maximum height `maxY` of
+  `0`,
+- We will then iterate these values - on each iteration, we will apply
```

Comment:
```suggestion
- We then iterate these values - on each iteration, we apply
```
Avoid the future tense here. There are more "will"'s down below, which should also be removed.